### PR TITLE
[PR FIX] Fixing the CI workflow 

### PR DESCRIPTION
**Pull request by CarlaShawi**
**Description:**

Adjusted the workflow so that Prettier first formats the code to ensure consistent styling. Once the files are formatted, Super-Linter checks the code for other issues, such as syntax errors, adherence to coding standards, and potential problems in JavaScript

Resolves Issue: #99

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -42,28 +42,30 @@ jobs:
       - name: Prettier Format
         uses: creyD/prettier_action@v4.3
         env:
-            GITHUB_TOKEN: ${{ secrets.TOKEN }} 
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (NicholasWahome) [inline .github/workflows/CI.yml]: In our repository it is secrets.TOKEN 
As the name in the repo is TOKEN only

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -42,28 +42,30 @@ jobs:
...
-            VALIDATE: "javascript"  # Only validate JavaScript
-            VALIDATE_ALL_CODEBASE: false
-            VALIDATE_MARKDOWN: false
-            DEFAULT_BRANCH: main 
-            GITHUB_TOKEN: ${{ secrets.TOKEN }} 
-            RUN_LOCAL: false
-            OUTPUT_DETAILS: 'minimal'  # Use minimal output
-            EXCLUDE_PATHS: "/github/workspace/client/node_modules/*,/github/workspace/server/node_modules/*"
+      - name: Run Super-Linter
+        uses: github/super-linter@latest
```
Comment (NicholasWahome) [inline .github/workflows/CI.yml]: Not sure why the change to this version as it has bugs and is not as stable
```suggestion
        uses: github/super-linter@latest
```

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -42,28 +42,30 @@ jobs:
...
+        env:
+          VALIDATE: "javascript" 
+          VALIDATE_ALL_CODEBASE: false  # Lint only the changed files, not the entire codebase
+          VALIDATE_MARKDOWN: false
+          IGNORE_FORMATTING: true  # Ignore formatting since Prettier handles it
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (NicholasWahome) [inline .github/workflows/CI.yml]: Again should be secrets.TOKEN

**Review by NicholasWahome — CHANGES_REQUESTED** [review]
Hello, please fix the following changes and it can then be merged into the main branch. 
Thank you @CarlaShawi

**Review by harismah — APPROVED** [review]
Ran the updated CI workflow locally, Prettify and Superlinter checked out and are checking the code appropriately after installing any dependencies.

**Review by harismah — APPROVED** [review]
Issues mentioned by @NicholasWahome seem to be resolved.

**Comment by NicholasWahome** [comment]
This looks perfect! @CarlaShawi

**Review by NicholasWahome — APPROVED** [review]
